tools/cargo_bazel/src/utils/starlark/label.rs

```diff
@@ -65,6 +66,39 @@ impl Display for Label {
     }
 }
 
+impl Label {
+    pub fn from_absolute_path(p: &Path) -> Result<Self, anyhow::Error> {
+        let mut workspace_root = None;
+        let mut package_root = None;
+        for ancestor in p.ancestors().skip(1) {
+            if package_root.is_none() && (ancestor.join("BUILD").exists() || ancestor.join("BUILD.bazel").exists()) {
+                package_root = Some(ancestor);
+            }
+            if workspace_root.is_none() && (ancestor.join("WORKSPACE").exists() || ancestor.join("WORKSPACE.bazel").exists()) {
+                workspace_root = Some(ancestor);
+                break;
+            }
+        }
+        match (workspace_root, package_root) {
```

Comment:
Nit, can you add a condition here that checks if there is no `package_root` and suggest to users that they need to add a `BUILD.bazel` file somewhere? That's what that case would mean, right?